config/peerpods/podvm/azure-podvm-image-handler.sh

```diff
@@ -504,15 +530,127 @@ function create_image() {
     # Create Azure image using packer
     create_image_using_packer
 
-    # Get the image id of the newly created image.
-    # This will set the IMAGE_ID variable
-    get_image_id
+    echo "Azure image created successfully from scratch"
+}
 
-    # Add the image id as annotation to peer-pods-cm configmap
-    add_image_id_annotation_to_peer_pods_cm
+# Function to create the azure image from prebuilt artifact
+# The prebuilt artifact is expected to be a vhd image
 
-    echo "Azure image created successfully"
+function create_azure_image_from_prebuilt_artifact() {
+    echo "Creating Azure image from prebuilt artifact"
+
+    # Set the IMAGE_VERSION and IMAGE_NAME
+    set_image_version_and_name
+
+    echo "Pulling the podvm image from the provided path"
+    IMAGE_SRC="/tmp/image"
+    EXTRACTION_DESTINATION_PATH="/image"
+    IMAGE_REPO_AUTH_FILE="/tmp/regauth/auth.json"
+
+    # Get the PODVM_IMAGE_TUPE and PODVM_IMAGE_SRC_PATH
+    get_image_type_url_and_path
+
+    case "${PODVM_IMAGE_TYPE}" in
+    oci)
+        echo "Extracting the Azure image from the given path."
+
+        mkdir -p "${EXTRACTION_DESTINATION_PATH}" ||
+            error_exit "Failed to create the image directory"
+
+        extract_container_image "${PODVM_IMAGE_URL}" "${PODVM_IMAGE_TAG}" "${IMAGE_SRC}" "${EXTRACTION_DESTINATION_PATH}" "${IMAGE_REPO_AUTH_FILE}"
+
+        # Form the path of the podvm raw image.
+        PODVM_IMAGE_PATH="${EXTRACTION_DESTINATION_PATH}/rootfs${PODVM_IMAGE_SRC_PATH}"
```

Comment:
i wonder if using "/rootfs/" would help to avoid wrong usage here

Reply:
I think the <dir>/rootfs is where the podvm image is available once extracted via umoci.

Reply:
i meant only to the additional "/"